powersimdata/network/hifld/model.py

```diff
@@ -0,0 +1,119 @@
+import os
+
+from powersimdata.input.abstract_grid import AbstractGrid
+from powersimdata.input.helpers import (
+    add_coord_to_grid_data_frames,
+    add_zone_to_grid_data_frames,
+    csv_to_data_frame,
+)
+from powersimdata.network.csv_reader import CSVReader
+from powersimdata.network.usa_tamu.constants.storage import defaults
+
+
+class HIFLD(AbstractGrid):
+    def __init__(self, interconnect):
```

Comment:
I think we could move all the instance methods to `AbstractGrid`, then just declare:
```
class HIFLD(AbstractGrid):
    pass
```
Same for the TAMU grid

Reply:
It makes sense. Is that still an abstract class then. It would look like a Parent class. Do we need to rename it?

Reply:
Oops, didn't see this one so replied below

Reply:
Could we do within the child classes:
```python
def __init__(self):
    self.top_dirname = os.path.dirname(__file__)
    super().__init__()
```
and then within the parent class:
````python
def __init__(self):
    self._set_data_loc()
    self.interconnect = check_and_format_interconnect(interconnect)
    self._build_network()

def _set_data_loc(self):
    data_loc = os.path.join(self.top_dirname, "data")
    if os.path.isdir(data_loc) is False:
        raise IOError("%s directory not found" % data_loc)
    else:
        self.data_loc = data_loc
```
?
So the child inits set the directory based on their `__file__`s, and the parent init uses the already-assigned directory within `_set_data_loc`.